napari/_qt/_qplugins/_qnpe2.py

```diff
@@ -0,0 +1,212 @@
+"""Plugin related functions that require Qt.
+
+Non-Qt plugin functions can be found in: `napari/plugins/_npe2.py`
+"""
+from __future__ import annotations
+
+import inspect
+from typing import (
+    TYPE_CHECKING,
+    List,
+    NamedTuple,
+    Optional,
+    Tuple,
+    Union,
+    cast,
+)
+
+from app_model import Action
+from app_model.types import SubmenuItem, ToggleRule
+from magicgui.type_map._magicgui import MagicFactory
+from magicgui.widgets import FunctionGui, Widget
+from npe2 import plugin_manager as pm
+from qtpy.QtWidgets import QWidget
+
+from napari._app_model import get_app
+from napari._app_model.constants import MenuGroup, MenuId
+from napari._qt._qapp_model.qactions._qproviders import _provide_window
+from napari.plugins import menu_item_template
+from napari.plugins._npe2 import (
+    _get_contrib_parent_menu,
+    get_widget_contribution,
+)
+from napari.utils.translations import trans
+from napari.viewer import Viewer
+
+if TYPE_CHECKING:
+    from npe2.manifest import PluginManifest
+    from npe2.plugin_manager import PluginName
+    from npe2.types import WidgetCreator
+
+    from napari._qt.qt_main_window import Window
+
+
+def _get_widget_viewer_param(
+    widget_callable: WidgetCreator, widget_name: str
+) -> str:
+    """Get widget parameter name (if any) and check type."""
```

Comment:
```suggestion
    """Get widget parameter name for `viewer` (if any) and check type."""
```

Reply:
Hmm @lucyleeow is this function doing double duty of checking if there's a `viewer` being requested **and** checking the widget type? If so, we should probably update the docstring a bit.